source_gen/lib/src/library.dart

```diff
@@ -26,26 +22,27 @@ class AnnotatedElement {
 class LibraryReader {
   final LibraryElement element;
 
-  Namespace _namespaceCache;
-
   LibraryReader(this.element);
 
-  Namespace get _namespace => _namespaceCache ??=
-      NamespaceBuilder().createExportNamespaceForLibrary(element);
-
   /// Returns a top-level [ClassElement] publicly visible in by [name].
   ///
   /// Unlike [LibraryElement.getType], this also correctly traverses identifiers
   /// that are accessible via one or more `export` directives.
-  ClassElement findType(String name) =>
-      element.getType(name) ?? _namespace.get(name) as ClassElement;
+  ClassElement findType(String name) {
+    var type = element.exportNamespace.get(name);
+    return type is ClassElement ? type : null;
+  }
 
   /// All of the declarations in this library.
   Iterable<Element> get allElements sync* {
     for (var cu in element.units) {
-      for (var compUnitMember in cu.unit.declarations) {
-        yield* _getElements(compUnitMember);
-      }
+      yield* cu.accessors;
```

Comment:
It'd be GREAT if analyzer had an API for this ( CC @scheglov @bwilkerson ) – could we get a TODO in here w/ a feature request?

These are places where things fall apart for weird reasons if things are reclassified, etc...